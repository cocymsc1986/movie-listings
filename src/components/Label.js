import React from 'react';

const Label = ({ label }) => {
	return (
		<div className="movies__genre-label">{label}</div>
	)
};

export default Label;