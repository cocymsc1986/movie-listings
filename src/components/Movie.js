import React, { Component } from 'react';
import Label from './Label';

export class Movie extends Component {

	render() {
		const { genreLabels, imagePath, title } = this.props;

		return (
			<article className="movies__result">
				<div className="movies__image"><img src={`https://image.tmdb.org/t/p/w200/${imagePath}`} alt={`${title}`} /></div>
				<div>
					<div className="movies__title"><h3>{title}</h3></div>
					<div className="movies__genre-labels">{genreLabels.map((label, key) => <Label label={label} key={key} />)}</div>
				</div>
			</article>
		);
	}
}

export default Movie;