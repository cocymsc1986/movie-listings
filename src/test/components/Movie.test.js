import React from 'react';
import Movie from '../../components/Movie';
import Label from '../../components/Label';

describe.only('Movie component', () => {
	let wrapper;

	beforeEach(() => {
		const props = {
			imagePath: 'path/to/image',
			title: 'test-title',
			movieKey: 1,
			genreLabels: ['test']
		};

		wrapper = shallow(<Movie {...props} />);
	});

	test('Should render', () => {
		expect(wrapper).toMatchSnapshot();
	});

	test('Should render label with correct prop', () => {
		expect(wrapper.find(Label).prop('label')).toBe('test');
	});
});