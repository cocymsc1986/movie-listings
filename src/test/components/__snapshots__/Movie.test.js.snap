// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Movie component Should render 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Movie
    genreLabels={
        Array [
            "test",
          ]
    }
    imagePath="path/to/image"
    movieKey={1}
    title="test-title"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="movies__image"
>
          <img
                    alt="test-title"
                    src="https://image.tmdb.org/t/p/w200/path/to/image"
          />
</div>,
        <div>
          <div
                    className="movies__title"
          >
                    <h3>
                              test-title
                    </h3>
          </div>
          <div
                    className="movies__genre-labels"
          >
                    <Label
                              label="test"
                    />
          </div>
</div>,
      ],
      "className": "movies__result",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <img
            alt="test-title"
            src="https://image.tmdb.org/t/p/w200/path/to/image"
/>,
          "className": "movies__image",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "alt": "test-title",
            "src": "https://image.tmdb.org/t/p/w200/path/to/image",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="movies__title"
>
              <h3>
                            test-title
              </h3>
</div>,
            <div
              className="movies__genre-labels"
>
              <Label
                            label="test"
              />
</div>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <h3>
                test-title
</h3>,
              "className": "movies__title",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "test-title",
              },
              "ref": null,
              "rendered": "test-title",
              "type": "h3",
            },
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <Label
                  label="test"
/>,
              ],
              "className": "movies__genre-labels",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": "0",
                "nodeType": "function",
                "props": Object {
                  "label": "test",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
            ],
            "type": "div",
          },
        ],
        "type": "div",
      },
    ],
    "type": "article",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="movies__image"
>
            <img
                        alt="test-title"
                        src="https://image.tmdb.org/t/p/w200/path/to/image"
            />
</div>,
          <div>
            <div
                        className="movies__title"
            >
                        <h3>
                                    test-title
                        </h3>
            </div>
            <div
                        className="movies__genre-labels"
            >
                        <Label
                                    label="test"
                        />
            </div>
</div>,
        ],
        "className": "movies__result",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <img
              alt="test-title"
              src="https://image.tmdb.org/t/p/w200/path/to/image"
/>,
            "className": "movies__image",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "alt": "test-title",
              "src": "https://image.tmdb.org/t/p/w200/path/to/image",
            },
            "ref": null,
            "rendered": null,
            "type": "img",
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="movies__title"
>
                <h3>
                                test-title
                </h3>
</div>,
              <div
                className="movies__genre-labels"
>
                <Label
                                label="test"
                />
</div>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <h3>
                  test-title
</h3>,
                "className": "movies__title",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "test-title",
                },
                "ref": null,
                "rendered": "test-title",
                "type": "h3",
              },
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <Label
                    label="test"
/>,
                ],
                "className": "movies__genre-labels",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": "0",
                  "nodeType": "function",
                  "props": Object {
                    "label": "test",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
              ],
              "type": "div",
            },
          ],
          "type": "div",
        },
      ],
      "type": "article",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getSnapshotBeforeUpdate": true,
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
